src/math/noise.js

```diff
@@ -198,8 +198,8 @@ let perlin; // will be initialized lazily by noise() or noiseSeed()
  */
 
 p5.prototype.noise = function(x, y = 0, z = 0) {
-  if (perlin == null) {
-    perlin = new Array(PERLIN_SIZE + 1);
+  if (perlin === null) {
```

Comment:
This causes the test to fail. `perlin` is never `null` but rather it is `undefined` at the start, the previous comparison is more loose so it passed.